src/browser/tab/mod.rs

```diff
@@ -58,6 +52,31 @@ pub type ResponseHandler = Box<
     + Sync,
 >;
 
+pub trait RequestInterceptor {
+    fn intercept(
+        &self,
+        transport: Arc<Transport>,
+        session_id: SessionId,
+        event: RequestPausedEvent,
+    ) -> RequestPausedDecision;
+}
+
+impl<
+        F: Fn(Arc<Transport>, SessionId, RequestPausedEvent) -> RequestPausedDecision + Send + Sync,
```

Comment:
super small nit but you may want to add this in a `where` clause for readability